src/cart/src/cart/crt_group.c

```diff
@@ -1638,45 +1539,33 @@ crt_hdlr_uri_lookup(crt_rpc_t *rpc_req)
 		D_GOTO(out, rc);
 	}
 	ul_out->ul_uri = cached_uri;
+	ul_out->ul_tag = ul_in->ul_tag;
+
 	if (ul_out->ul_uri != NULL)
 		D_GOTO(out, rc);
 
+	/* If this server does not know rank:0 then return error */
+	if (ul_in->ul_tag == 0)
+		D_GOTO(out, rc = -DER_OOG);
+
 	/**
-	 * step 2, make sure URI of tag 0 is in cache in preparation to forward
-	 * the lookup RPC
+	 * step 2, if rank:tag is not found, lookup rank:tag=0
 	 */
-	if (ul_in->ul_tag != 0) {
-		/* tag != 0, check if tag 0 is in cache */
-		rc = crt_grp_lc_lookup(grp_priv_primary, crt_ctx->cc_idx,
+	ul_out->ul_tag = 0;
+	rc = crt_grp_lc_lookup(grp_priv_primary, crt_ctx->cc_idx,
```

Comment:
Alignment?

Reply:
hm not sure what alignment you are referring to. it appears properly aligned within scope of block to me

Reply:
The line below, but github wouldn't allow me to comment on that as it was unmodified.

Reply:
ah ok, will change